lib/Login.dart

```diff
@@ -1,27 +1,95 @@
+import 'dart:convert';
 import 'package:flutter/material.dart';
 import 'Home.dart';
+import 'main.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+
+GoogleSignIn googleSignIn = GoogleSignIn(
+  scopes: [
+    'email',
+    'https://www.googleapis.com/auth/userinfo.profile',
+  ],
+);
+
+_handleSignIn() async {
+  try {
+    await googleSignIn.signIn();
+  } catch (error) {
+    print(error);
+  }
+}
+
+Future<String> tokenFromAccount(GoogleSignInAccount account) async {
+  GoogleSignInAuthentication auth;
+  try {
+    auth = await account.authentication;
+    print('okay');
+  } catch (error) {
+    print('error');
+  }
+  return auth.idToken;
+}
 
 class Login extends StatefulWidget {
   @override
   _LoginState createState() => _LoginState();
 }
 
 class _LoginState extends State<Login> {
+  GoogleSignInAccount currentUser;
+  bool success;
+
+  setCurrentUser(GoogleSignInAccount account) {
+    setState(() {
+      currentUser = account;
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    currentUser = null;
+    success = false;
+    try {
+      googleSignIn.signInSilently(); }
+    catch (error) {
+      googleSignIn.signIn();
+    }
+    googleSignIn.onCurrentUserChanged.listen((GoogleSignInAccount account) {
+      setCurrentUser(account);
+      tokenFromAccount(currentUser).then((token) async {
+        return await authenticationRequest(token);
+      }).then((response) {
+        var json = jsonDecode(response);
+        setState(() {
+          success = json['success'] == 'true' ? true : false;
```

Comment:
Looks like it should just be `success = json['success'] == 'true'`

Reply:
Not sure whether it will ever work. When the promise resolves, the method may already finished execution, so you are assigning to a local variable that no longer in scope.